### PR TITLE
Management UI DS and SM metadata mgmt

### PR DESCRIPTION
**Pull request by kaizimmerm**
Management UI handles Software Module and Distribution Set Metadata.

This has been in Management API for quite some time but was still missing in UI.

### CODE REVIEW

File: hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/common/AbstractMetadataPopupLayout.java [hunk 1]
```diff
@@ -0,0 +1,492 @@
...
+        metadataWindow.setHeight(550, Unit.PIXELS);
+        metadataWindow.setWidth(800, Unit.PIXELS);
+        metadataWindow.getMainLayout().setSizeFull();
+        metadataWindow.setResizable(true);
+        metadataWindow.addWindowModeChangeListener(event -> onResize(event));
+        setUpDetails(entity.getId(),metaData);
+        return metadataWindow;
+    }
+
+    public void setUpDetails(final Long swId, final M metaData) {
```
Comment (Jonkno) [inline hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/common/AbstractMetadataPopupLayout.java]: Javadoc

File: hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/distributions/dstable/DsMetadataPopupLayout.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    private EntityFactory entityFactory;
+
+    @Override
+    protected void checkForDuplicate(DistributionSet entity, String value) {
+        distributionSetManagement.findOne(entity, value);
+    }
+    /**
+     * Create metadata for DistributionSet.
+     */
+
```
Comment (Jonkno) [inline hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/distributions/dstable/DsMetadataPopupLayout.java]: Remove empty line

File: hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/distributions/event/SwModuleUIEvent.java [hunk 3]
```diff
@@ -13,6 +13,6 @@
  *
  */
 public enum SwModuleUIEvent {
```
Comment (Jonkno) [inline hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/distributions/event/SwModuleUIEvent.java]: Missing javadoc

File: hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/common/AbstractMetadataPopupLayout.java [hunk 4]
```diff
@@ -0,0 +1,492 @@
...
+        swcontactContainer.addContainerProperty(VALUE, String.class, "");
+        swcontactContainer.addContainerProperty(DELETE_BUTTON, String.class, FontAwesome.TRASH_O.getHtml());
+        return swcontactContainer;
+    }
+
+    private void popualateKeyValue(final Object metadataCompositeKey) {
+        if (metadataCompositeKey != null) {
+            Item item = metaDataGrid.getContainerDataSource().getItem(metadataCompositeKey);
+            keyTextField.setValue((String) item.getItemProperty(KEY).getValue());
+            // TODO value to be stored in table???
```
Comment (schabdo) [inline hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/common/AbstractMetadataPopupLayout.java]: Found a `TODO`

**Comment by kaizimmerm** [comment]
Closed due to broken commit history

**Pull request closed without merging**